pkg/chunkenc/memchunk.go

```diff
@@ -497,11 +502,38 @@ func (c *MemChunk) Iterator(ctx context.Context, mintT, maxtT time.Time, directi
 	return iter.NewEntryReversedIter(iterForward)
 }
 
-func (b block) iterator(ctx context.Context, pool ReaderPool, filter logql.LineFilter) iter.EntryIterator {
+// Blocks implements Chunk
+func (c *MemChunk) Blocks(mintT, maxtT time.Time) []Block {
+	mint, maxt := mintT.UnixNano(), maxtT.UnixNano()
+	blocks := make([]Block, 0, len(c.blocks))
+
+	for _, b := range c.blocks {
+		if maxt > b.mint && b.maxt > mint {
+			blocks = append(blocks, b)
+		}
+	}
+	return blocks
```

Comment:
could we gain anything by slicing up the existing c.blocks instead of allocating a new slice?  Also curious if c.blocks was a slice of pointers if we could save a copy of the block here?

Reply:
I’ll check if it does help.  There’s other place where I don’t reslice intentionally because reslicing keep underlying references.